src/foam/nanos/auth/ServiceProviderAware.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.INTERFACE({
+  package: 'foam.nanos.auth',
+  name: 'ServiceProviderAware',
+
+  methods: [
```

Comment:
Other interfaces I'd consider very similar to this one, like `LastModifiedAware`, `CreatedByAware`, etc. just define a property rather than getter and setter methods. Wouldn't we want `ServiceProviderAware` to be consistent with the other ones? Or is there a reason we want this to be different?

Reply:
It's different because it already is different.  ServiceProvider.js exists and User already was using spid. 
And I don't understand the 'just add a property' - this is an interface. Interfaces don't support properties.

Reply:
> And I don't understand the 'just add a property' - this is an interface. Interfaces don't support properties.

Interfaces don't in general in the OOP sense, but they appear to in FOAM. Which makes them more like mix-ins or traits than an interface, but it's still a `foam.INTERFACE`.

See the definition of `CreatedByAware` (and the other ones I mentioned above):

```Java
foam.INTERFACE({
  package: 'foam.nanos.auth',
  name: 'CreatedByAware',

  properties: [
    {
      class: 'Reference',
      of: 'foam.nanos.auth.User',
      name: 'createdBy',
      documentation: 'User who created the entry'
    }
  ]
});
```

Although there's a bug right now where saying a model implements one of these traits/mix-ins/interfaces doesn't actually add the property to the model, that model still has to add the property to the list "manually".

Reply:
Well that's handy. Thanks.